types/uint.go

```diff
@@ -172,3 +180,43 @@ func checkNewUint(i *big.Int) (Uint, error) {
 	}
 	return Uint{i}, nil
 }
+
+
+// RelativePow raises x to the power of n, where x (and the result, z) are scaled by factor b
+// for example, RelativePow(210, 2, 100) = 441 (2.1^2 = 4.41)
+func RelativePow(x Uint, n Uint, b Uint) (z Uint) {
+	if x.IsZero() {
+		if n.IsZero() {
+			z = b // 0^0 = 1
```

Comment:
comment makes sense, but we just set z to the base `b`.

Reply:
It's to preserve the scaling factor. z (the result), is scaled by b, so returning b will give z/b = 1.